### ISSUE TITLE
DateField: formatting of selected date ignores Locale

### ISSUE THREAD

**Issue by mathisdt** [vaadin/framework#10740]
When I set the Locale globally in Vaadin and/or for a DateField locally, the dropdown is correctly localized but when I select a date, the formatting doesn't use my Locale but prints English weekday names. I would expect that the formatted date was also localized using the provided Locale.

## example (also see attached example project [datefield-bugdemo.zip](https://github.com/vaadin/framework/files/1837245/datefield-bugdemo.zip))

The code
`final DateField dateField = new DateField();`
`dateField.setLocale(new Locale("de", "DE"));`
`dateField.setDateFormat("EE, dd.MM.yyyy");`
displays the dropdown correctly localized:
![date-picker-open](https://user-images.githubusercontent.com/1932462/37767718-5f648506-2dcb-11e8-8105-c726602b4d10.png)
but when selected, the date is not localized anymore:
![date-selected](https://user-images.githubusercontent.com/1932462/37767719-5f805b64-2dcb-11e8-88ac-5f4b07a8448a.png)

## used versions
- Vaadin Framework 8.3.2
- Firefox 59.0.1
- Spring Boot 2.0.0

**Comment by TatuLund** [comment]
Is this duplicate of this one https://github.com/vaadin/framework/issues/9765 ?

**Comment by mathisdt** [comment]
No, that issue seems unrelated. This issue is purely about localization of date parts (e.g. weekdays), not about specific dates which are perhaps displayed incorrectly in the whole component.

**Comment by mathisdt** [comment]
It seems that the test cases in #10188 only verify that the week days **inside the popup** are correct. This ticket is about the week day displayed in the field **after the popup is closed** (after a date was selected).

**Comment by TatuLund** [comment]
One source of the problems is actually this. I.e. the conversion method used in GWT does support time zone only to one direction.

https://code.google.com/archive/p/google-web-toolkit/issues/5868
